pkg/services/ngalert/notifier/receiver_svc.go

```diff
@@ -61,19 +63,71 @@ func NewReceiverService(
 	}
 }
 
-func (rs *ReceiverService) canDecrypt(ctx context.Context, user identity.Requester, name string) (bool, error) {
-	receiverAccess := false // TODO: stub, check for read secrets access
-	eval := accesscontrol.EvalPermission(accesscontrol.ActionAlertingProvisioningReadSecrets)
-	provisioningAccess, err := rs.ac.Evaluate(ctx, user, eval)
+func (rs *ReceiverService) shouldDecrypt(ctx context.Context, user identity.Requester, name string, reqDecrypt bool) (bool, error) {
+	// TODO: migrate to new permission
+	eval := accesscontrol.EvalAny(
+		accesscontrol.EvalPermission(accesscontrol.ActionAlertingReceiversReadSecrets),
+		accesscontrol.EvalPermission(accesscontrol.ActionAlertingProvisioningReadSecrets),
+	)
+
+	decryptAccess, err := rs.ac.Evaluate(ctx, user, eval)
 	if err != nil {
 		return false, err
 	}
-	return receiverAccess || provisioningAccess, nil
+
+	if reqDecrypt && !decryptAccess {
+		return false, ErrPermissionDenied
+	}
+
+	return decryptAccess && reqDecrypt, nil
+}
+
+// GetReceiver returns a receiver by name.
+// The receiver's secure settings are decrypted if requested and the user has access to do so.
+func (rs *ReceiverService) GetReceiver(ctx context.Context, q models.GetReceiverQuery, user identity.Requester) (definitions.GettableApiReceiver, error) {
```

Comment:
I see some authz code in GetReceivers but not here. What is the reason of omitting here and adding there?

Reply:
I don't think the single read action should care about `:list`, there should never be a case where you call this method without `:read`. At the API level you need `:read` access to call the single GET endpoint, but we could codify that here. WDYT?